clients/netcore/APIAuthHandler.cs

```diff
@@ -54,12 +54,16 @@ public APIAuthHandler(string authUrl, string scope, string clientId, string clie
 		}
 
 		protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) {
+
 			if (DateTime.Now.Ticks > _tokenValidUntil.Ticks) {
-				lock(_lockobj) {
+
+				await _refreshTokenLock.WaitAsync();
```

Comment:
We could/should pass `cancellationToken` to `WaitAsync`. It's also okay for me if we integrate it after 1.9.0